engine/table/src/main/java/io/deephaven/engine/table/impl/lang/QueryLanguageParser.java

```diff
@@ -2430,6 +2459,56 @@ public Class<?> visit(MethodCallExpr n, VisitArgs printer) {
                 typeArguments);
     }
 
+    private Optional<CastExpr> makeCastExpressionForPyCallable(Class<?> retType, MethodCallExpr callMethodCall) {
+        if (retType.isPrimitive()) {
+            return Optional.of(new CastExpr(
+                    new PrimitiveType(PrimitiveType.Primitive
+                            .valueOf(retType.getSimpleName().toUpperCase())),
+                    callMethodCall));
+        } else if (retType.getComponentType() != null) {
+            Class<?> componentType = retType.getComponentType();
+            if (componentType.isPrimitive()) {
+                ArrayType arrayType;
+                if (componentType == boolean.class) {
+                    arrayType = new ArrayType(new ClassOrInterfaceType(null, "java.lang.Boolean"));
```

Comment:
Should this use `StaticJavaParser.parseClassOrInterfaceType`? Same for other usages below.